llmfoundry/models/mpt/modeling_mpt.py

```diff
@@ -422,6 +422,7 @@ def forward(
         )
 
         # initialize the past key values cache if it should be used
+        presents = () if use_cache else None
         if use_cache and past_key_values is None:
             past_key_values = [() for _ in range(self.config.n_layers)
```

Comment:
typing nit:
past_key_values is a list
presents is a tuple

Reply:
BaseModelOutputWithPast expects a tuple of tuples, but past_key_values is expected to be a list of tuples 😅